x-pack/plugins/observability/public/utils/slo/convert_sli_apm_params_to_apm_app_deeplink_url.ts

```diff
@@ -48,7 +48,7 @@ export function convertSliApmParamsToApmAppDeeplinkUrl(slo: SLOResponse): string
   if (transactionName && transactionName !== ALL_VALUE) {
     kueryParams.push(`transaction.name : "${transactionName}"`);
   }
-  if (filter && filter.length > 0) {
+  if (filter && typeof filter === 'string' && filter.length > 0) {
```

Comment:
How to handle unified search filters when navigating to apm?

Reply:
not sure about it, for this we will have to support unified search in APM first